src/DataLib/fendl32B_retrofit/tendl_processing.py

```diff
@@ -1,6 +1,64 @@
 # Import packages
 import ENDFtk
-import pandas as pd
+from pathlib import Path
+
+def get_isotope(stem):
+    """
+    Extract the element name and mass number from a given filename.
+
+    Arguments:
+        stem (str): Stem of a an ENDF (TENDL) and/or PENDF file, formatted
+            as f'{element}{mass_number}.ext'
+    Returns:
+        element (str): Chemical symbol of the isotope whose data is contained
+            in the file.
+        A (str): Mass number of the isotope, potentially including the letter
+            "m" at the end if the isotope is in a metastable state.
+    """
+
+    for i, char in enumerate(stem):
+        if char.isdigit():
+            break
+    
+    element = stem[:i]
+    A = stem[i:]       
+    
+    return element, A
+
+def search_for_files(dir = '.'):
+    """
+    Search through a directory for all pairs of ENDF (TENDL) and PENDF files
+        that have matching stems. If so, save the paths and the isotopic
+        information to a dictionary.
+    
+    Arguments:
+        directory (str, optional): Path to the directory in which to search
+            for ENDF and PENDF files.
+            Defaults to the present working directory (".").
+    Returns:
+        file_info (dict): Dictionary containing the chemical symbol, mass
+            number, and paths to the ENDF and PENDF files for a given isotope.
+            The dictionary is formatted as such:
+            {f'{element}{mass_number}' :
+                                    {'Element'} : Isotope's chemical symbol,
+                                    {'Mass Number'} : Isotope's mass number,
+                                    {'File Paths'} : (endf_path, pendf_path)
+            }
+    """
+
+    dir = Path(dir)
+
+    file_info = {}
+    for file in (p for p in dir.glob('*') if p.suffix in {'.tendl', '.endf'}):
+        if file.is_file() and file.with_suffix('.pendf').is_file():
+            element, A = get_isotope(file.stem)
+            file_info[f'{element}{A}'] = {
+                'Element'       :                              element,
+                'Mass Number'   :                                    A,
+                'File Paths'    :   (file, file.with_suffix('.pendf'))
+            }
```

Comment:
I was thinking that the glob could allow you to be more selective about which files you are even looking for:
```suggestion
    for suffix in ['tendl', 'endf']:
        for file in dir.glob(f'*.{suffix}'):
            if file.with_suffix('.pendf').is_file():
                element, A = get_isotope(file.stem)
                file_info[f'{element}{A}'] = {
                    'Element'       :                              element,
                    'Mass Number'   :                                    A,
                    'File Paths'    :   (file, file.with_suffix('.pendf'))
                }
```